### ISSUE TITLE
Staging prostředí pro materiály

### ISSUE THREAD

**Issue by FilipJirsak** [podlomar/kodim.cz#62]
Hodilo by se mít staging prostředí pro publikování materiálů. Push do main větve by vypublikoval změny na staging prostředí, kde by se to dalo zkontrolovat, a teprve pak by se mergem do jiného branche udělal deploy na hlavní server.

**Comment by FilipChalupa** [comment]
> a teprve pak by se mergem do jiného branche udělal deploy na hlavní server

Jestli je to mergem nebo jinak asi kodim nemusí řešit, ne?

Prostě push do `main` nasadí stav na mainu do `//stage.kodim.cz` a push do `deploy` nasadí stav deploy větve do `//kodim.cz`. Nebo co prosím ten merge řeší?

Z mého pohledu deploy lokálního headu na stage i produkci by mohl být vynucen třeba takto:

```sh
git push origin HEAD:main
git push origin HEAD:deploy
```

nebo zkráceně

```sh
git push origin HEAD:main HEAD:deploy
```

**Comment by FilipJirsak** [comment]
Ano, kodim.cz by neřešilo, jak se commit do hlavní větve dostal. Popisoval jsem běžný postup pomocí pull requestů, ale jestli by to bylo vynucené nebo ne záleží na konfiguraci repository, tedy jestli by hlavní větev byla chráněná.

**Comment by FilipChalupa** [comment]
Ok
